test/functional/api/vim_spec.lua

```diff
@@ -3620,6 +3629,12 @@ describe('API', function()
         msg_a{15:msg_b}{16:msg_c}                         |
       ]],
       }
+      async_meths.nvim_echo({
+        { 'msg_a' },
+        { 'msg_b', api.nvim_get_hl_id_by_name('Statement') },
+        { 'msg_c', api.nvim_get_hl_id_by_name('Special') },
+      }, true, {})
+      screen:expect_unchanged()
```

Comment:
nit: using `screen:expect_unchanged()` when a `screen:expect()` works fine (just needing some minor change) is considered an anti-pattern, as is it less robust than expecting a different state. 

In this case, better to change the messages slightly (like using `msg_d` or whatever), and then assert this distinct state.

Reply:
I e `screen:unchanged()` is more for asserting that nothing happened at all UI-wise, which is not the case here. we are actively interested in a _different_ message.

Reply:
Hmm yeah I knew it was intended to be used like that but I've also used it (here and in statuslcolumn/sign tests) as a way to test that different commands yield the same screen state. Adding new `screen:expect()` in such cases usually leads to warnings.